tests/kafkatest/services/kafka/kafka.py

```diff
@@ -91,12 +116,11 @@ def stop_node(self, node, clean_shutdown=True):
 
         for pid in pids:
             node.account.signal(pid, sig, allow_fail=False)
-
-        node.account.ssh("rm -f /mnt/kafka.pid", allow_fail=False)
+        wait_until(lambda: len(self.pids(node)) == 0, timeout_sec=20, err_msg="Kafka node failed to stop")
 
     def clean_node(self, node):
         node.account.kill_process("kafka", clean_shutdown=False, allow_fail=True)
-        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log /mnt/kafka.pid", allow_fail=False)
+        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log", allow_fail=False)
```

Comment:
Are you sure that's what we want? If we go the mixin route for stuff like collecting JMX metrics, we may need to be careful about cleaning up like that -- there will be data from multiple classes in `/mnt`, so blindly removing all data may break other code depending on which order their clean up code is invoked.